subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/MultiProjectContinuousIntegrationTest.groovy

```diff
@@ -41,6 +44,14 @@ class MultiProjectContinuousIntegrationTest extends AbstractContinuousIntegratio
 
         upstreamSource = file("upstream/src/main/java/Upstream.java") << "class Upstream {}"
         downstreamSource = file("downstream/src/main/java/Downstream.java").createFile() << "class Downstream extends Upstream {}"
+
+        // MacOS builds with Intel machines can be too fast on CI and
+        // changes generated during build are picked up only after the build which retriggers the build
+        // Fix for issue: https://github.com/gradle/gradle-private/issues/3728
+        if (MAC_OS_X.fulfilled && NOT_MAC_OS_X_M1.fulfilled) {
+            def quietPeriod = 250
```

Comment:
💭 I suppose you checked if that quiet period is enough.

Reply:
Yes, I run it a few times, e.g.: https://builds.gradle.org/buildConfiguration/Gradle_Master_Util_RerunFlakyTestMacosAmd64/61673271?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true

https://builds.gradle.org/buildConfiguration/Gradle_Master_Util_RerunFlakyTestMacosAmd64/61673966?hideProblemsFromDependencies=false&hideTestsFromDependencies=false